rewrite-java-11/src/main/java/org/openrewrite/java/Java11Parser.java

```diff
@@ -75,6 +82,7 @@
     private final Collection<NamedStyles> styles;
 
     private Java11Parser(@Nullable Collection<Path> classpath,
+                         Collection<byte[]> classBytesClasspath,
```

Comment:
construction of the parser is expensive, we really don't want to create a new parser each time to inject dynamic classes into the classpath during construction. 

Rather, we should have an overloaded `parse` method that takes the collection of byte arrays and those are then converted to `JavaFileObject` instances and set on the parser.

We also need to make sure this same functionality exists on the Java 8 parser.

Reply:
@tkvangorder I think what was expensive about constructing the parser was when we supplied Java source stubs to the parser at construction time which required a semi-compilation phase. Supplying byte arrays as classpath elements requires no work at construction time, so I think we can start thinking of parser construction as cheap. I might be missing something.

Reply:
I guess if we changed this to use a builder, we could get rid of the need to use a thread local to hold the parser.